src/AzureExtension/Services/DevBox/DataTokenService.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using AzureExtension.Contracts;
+using DevHomeAzureExtension.DeveloperId;
+using Microsoft.Windows.DevHome.SDK;
+
+namespace AzureExtension.Services.DevBox;
+
+public class DataTokenService : IDataTokenService
+{
+    private AuthenticationHelper AuthHelper => new();
+
+    public async Task<string> GetTokenAsync(IDeveloperId? devId)
+    {
+        if (devId == null)

```

Comment:
Consider adding logging here to know that the developer id provided was null for potential diagnosis of issues